src/System.Text.Encoding/tests/EncodingTestHelpers.cs

```diff
@@ -23,7 +42,7 @@ public static void GetByteCount(Encoding encoding, string chars, int index, int
             Assert.Equal(expected, encoding.GetByteCount(charArray, index, count));
         }
         
-        public static void GetBytes(Encoding encoding, string source, int index, int count, byte[] bytes, int byteIndex, byte[] expectedBytes)
+        private static void GetBytes(Encoding encoding, string source, int index, int count, byte[] bytes, int byteIndex, byte[] expectedBytes)
```

Comment:
you code is allocating the result buffer and then passing this buffer to GetBytes. my points is, we don't have to allocate the buffer outside GetBytes. let GetBytes allocate the buffer and validate it as needed. imagine in the future we need to call GetBytes in other places, at that time we'll need to repeat the action of the buffer allocation and validation. in short, let GetBytes be self contained and let it handle getting the results and test it against the expected results. let me know if this is still unclear